### ISSUE TITLE
Add Container Security Scanning to CI/CD Process

### ISSUE THREAD

**Issue by lewisdaly** [mojaloop/project#1003]
## **Goal**:
As an OSS Maintainer
I want to _enable container security scans in the CI/CD Process_
so that _containers built and released by Mojaloop releases for services are secure_ and any security issues can be identified

**Tasks**:
- [x] Locally compare `sysdig` and `anchore-cli` tools
> It looks like sysdig and anchore-cli do quite different things (and can actually play quite nicely together). Sysdig is more for runtime container security management, while anchore-cli is for build time security scanning
- [x] Pending the above decision, integrate the chosen tool into the CI/CD process of the `account-lookup-service`
- [ ] Evaluate the above, and if we are happy, go ahead and integrate into the CI/CD process of the following:
  - [x] bulk-api-adapter [PR]
  - [x] central-event-processor [PR]
  - [x] central-ledger [PR]
  - [x] central-settlement [PR]
  - [x] email-notifier [PR]
  - [x] ml-api-adapter [PR]
  - [x] quoting-service [PR]
  - [x] mojaloop-simulator [PR]
  ~- [ ] central-directory?~ _Out of scope_
  ~- [ ] simulator?~ _Out of scope_

**Acceptance Criteria**:
- [x] Container scans are added across all codebases which produce Docker images
- [x] Appropriate reports are accessible in CircleCI artifacts

## **Pull Requests**:
- [x] https://github.com/mojaloop/account-lookup-service/pull/103
- [x] https://github.com/mojaloop/account-lookup-service/pull/165
- [x] https://github.com/mojaloop/central-event-processor/pull/89
- [x] https://github.com/mojaloop/bulk-api-adapter/pull/16
- [x] https://github.com/mojaloop/central-settlement/pull/195
- [x] https://github.com/mojaloop/central-ledger/pull/498
- [x] https://github.com/mojaloop/email-notifier/pull/85
- [x] https://github.com/mojaloop/ml-api-adapter/pull/291
- [x] https://github.com/mojaloop/quoting-service/pull/103
- [x] https://github.com/mojaloop/mojaloop-simulator/pull/22

## **Follow-up**:
- Expand this scanning capability to all currently deployed versions of the given mojaloop containers. This exists beyond the scope of an individual release.
- Add container scanning to the Helm releases?
- Create separate stories for issues identified during initial scans

**Dependencies**:
- N/A

## **Accountability**:
- Owner: @lewisdaly 
- QA/Review: TBC

**Comment by lewisdaly** [comment]
I've enabled the scans for a branch of the account-lookup-service:

https://circleci.com/gh/mojaloop/account-lookup-service/1596#artifacts/containers/0

There's a few issues we need to figure out, such as: 
- if `policy_failure` in the image-scan step is set to true, the build artifacts aren't generated

**Comment by lewisdaly** [comment]
Here's an example of the updated CI workflow:

<img width="1483" alt="Screen Shot 2019-10-10 at 2 41 07 pm" src="https://user-images.githubusercontent.com/1683980/66544919-0be40900-eb6c-11e9-8cd0-dbaf12503c94.png">

I've separated out the build, and publish steps, so that we can insert the docker image license-scan and image-scan in parallel here.

**Comment by elnyry** [comment]
@lewisdaly thanks for the updates..

Looking at the snapshot, is the container security scanning enabled only for releases/snapshots or even PRs?

**Comment by lewisdaly** [comment]
I have it set up only for releases/snapshots, but that is because historically, we only build a docker image for the release/snapshot tags.

Is that fine? Or we could change it to build a docker image on every branch/tag/PR, and only publish the docker images if we are building for a release/snapshot.

**Comment by lewisdaly** [comment]
I should note that the image scan is quite a long process (5 mins +), so putting it on every PR might get annoying.

**Comment by elnyry** [comment]
In fact that was what I was leaning towards as well - to have it only for snapshots / releases and not for PRs, since the build happens only for those anyway..

**Comment by rasputtintin** [comment]
This is great work. Can we look at:
1 - Policy Settings so we can configure our baselines
2 - Reporting / Alerts to allow us do scheduled scans and have reports emailed to relevant teams to re-mediate
3 - Explore ways of tracking issues over time so we can report on progress / trends.

**Comment by lewisdaly** [comment]
@rasputtintin here is a link to the default policy that is being applied:

https://github.com/anchore/hub/blob/master/sources/bundles/anchore_default_bundle.json

I'm not exactly sure how it works, but perhaps we can continue the integration on the other repos with just the default policy for now.

**Comment by lewisdaly** [comment]
@elnyry Marked `central-directory` and `simulator` as out of scope since I want to focus our attention on services that not be deprecated in the near future.
